sources/packages/backend/apps/workers/src/zeebe/zeebe-transport-strategy.ts

```diff
@@ -63,7 +64,15 @@ export class ZeebeTransportStrategy
       `Starting job for processInstanceKey ${job.processInstanceKey}. Retries left: ${job.retries}.`,
     );
     try {
-      return await jobHandler(job);
+      // The return is an `Observable`for a jobHandler, which needs to be awaits.
+      // The lastValueFrom is almost exactly the same as toPromise()
+      // meaning that it will resolve with the last value that has arrived
+      // when the Observable completes.
+      const jobResult = await jobHandler(job);
+      if (isObservable(jobResult)) {
+        await lastValueFrom(jobResult);
```

Comment:
I believe that the comment should focus more on the two possible results and why `await jobHandler(job);` can receive one or another instead of explaining the `lastValueFrom`.
As a suggestion:
"The jobHandler can potentially return a Promise or an Observable. The Promise will be returned from a controller when it finishes as expected returning some value. The Observable will be returned as a result of an unhandled exception when a RpcExecption will be generated and the RpcExpectionHandler will wrap this exception as an Observable object."